cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/SearchRequest.java

```diff
@@ -0,0 +1,196 @@
+      throw new IllegalArgumentException("offset must not be negative");
+    }
+    if (limit < 0) {
+      throw new IllegalArgumentException("limit must not be negative");
+    }
+    this.namespace = namespace;
+    this.query = query;
+    this.types = Collections.unmodifiableSet(new HashSet<>(types));
+    this.sortInfo = sortInfo;
+    this.numCursors = numCursors;
+    this.cursor = cursor;
+    this.showHidden = showHidden;
+    this.entityScope = Collections.unmodifiableSet(new HashSet<>(entityScope));
+    this.offset = offset;
+    this.limit = limit;
+  }
+
+  /**
+   * @return the namespace to search in
+   */
+  public NamespaceId getNamespace() {
+    return namespace;
+  }
+
+  /**
+   * The search query can be of two form: [key]:[value] or just [value] and can have '*' at the end for a prefix search.
+   *
+   * @return the search query
+   */
+  public String getQuery() {
+    return query;
+  }
+
+  /**
+   * The types of entities to restrict the search to. If empty, all types should be searched.
+   *
+   * @return the types of entities to restrict the search to
+   */
+  public Set<EntityTypeSimpleName> getTypes() {
+    return types;
+  }
+
+  /**
+   * @return how to sort the results
+   */
+  public SortInfo getSortInfo() {
+    return sortInfo;
+  }
+
+  /**
+   * The offset to start with in the search results. {@code 0} means results should be returned from the beginning.
+   * Only applies when {@link #getSortInfo()} is not {@link SortInfo#DEFAULT}.
+   *
+   * @return the offset to start the results at
+   */
+  public int getOffset() {
+    return offset;
+  }
+
+  /**
+   * The max number of results to return, starting from {@link #getOffset()}. Only applies when {@link #getSortInfo()}
+   * is not {@link SortInfo#DEFAULT}.
+   *
+   * @return the maximum number of results to return
+   */
+  public int getLimit() {
+    return limit;
+  }
+
+  /**
+   * The number of cursors to return in the response. A cursor identifies the first index of the next page for
+   * pagination purposes. Only applies when {@link #getSortInfo()} is not {@link SortInfo#DEFAULT}.
+   *
+   * @return number of cursors to return in the response
+   */
+  public int getNumCursors() {
+    return numCursors;
+  }
+
+  /**
+   * The cursor that acts as the starting index for the request page. This is only applicable when
+   * {@link #getSortInfo()} is not {@link SortInfo#DEFAULT}. If offset is also specified, it is applied starting at
+   * the cursor. If not present, the first row is used as the cursor.
+   *
+   * @return the cursor for search results, or null if there is no cursor
+   */
+  @Nullable
+  public String getCursor() {
+    return cursor;
+  }
+
+  /**
+   * @return whether to display hidden entities (entities whose name start with '_').
+   */
+  public boolean shouldShowHidden() {
+    return showHidden;
+  }
+
+  /**
+   * @return scope of entities to display
+   */
+  public Set<EntityScope> getEntityScope() {
```

Comment:
Defines whether the search is done on the system metadata table or user metadata table.  if a scope not defined the scan is performed on both and results are aggregated .